noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/tail_output_composer/meter_gas_used.nr

```diff
@@ -27,8 +26,12 @@ pub fn meter_gas_used(data: PrivateToRollupAccumulatedData) -> Gas {
     metered_da_fields += num_l2_to_l1_msgs;
     metered_l2_gas += num_l2_to_l1_msgs * L2_GAS_PER_L2_TO_L1_MSG;
 
+    // Note: not sure if we should be measuring gas in bytes since we use blobs?
     let num_private_logs = array_length(data.private_logs);
-    metered_da_fields += num_private_logs * PRIVATE_LOG_SIZE_IN_FIELDS;
+    /// Safety: the length below is constrained in rollup base. Should we constrain it here?
+    let total_num_private_log_fields =
+        unsafe { get_emitted_private_log_fields_len(data.private_logs) };
+    metered_da_fields += total_num_private_log_fields;
```

Comment:
I'm not sure whether we still want to measure overall logs in bytes? If so, how should metering gas be handled for tightly packed logs?